src/NServiceBus.AcceptanceTests/Routing/When_using_assembly_level_message_mapping_for_pub_sub.cs

```diff
@@ -0,0 +1,88 @@
+﻿namespace NServiceBus.AcceptanceTests.Routing
+{
+    using System.Threading.Tasks;
+    using AcceptanceTesting;
+    using AcceptanceTesting.Customization;
+    using EndpointTemplates;
+    using NServiceBus.Config;
+    using NUnit.Framework;
+    using ScenarioDescriptors;
+
+    public class When_using_assembly_level_message_mapping_for_pub_sub : NServiceBusAcceptanceTest
+    {
+        static string OtherEndpointName => Conventions.EndpointNamingConvention(typeof(OtherEndpoint));
+
+        [Test]
+        public async Task The_mapping_should_not_cause_publishing_to_non_subscribers()
+        {
+            await Scenario.Define<Context>()
+                .WithEndpoint<OtherEndpoint>()
+                .WithEndpoint<Publisher>(b =>
+                    b.When(c => c.EndpointsStarted, async session =>
+                    {
+                        await session.Publish(new MyEvent());
+                        await session.Send(new DoneCommand());
+                    })
+                )
+                .Done(c => c.Done)
+                .Repeat(r => r.For<AllTransportsWithMessageDrivenPubSub>())
+                .Run();
+        }
```

Comment:
The test blows up because the `OtherEndpoint` gets a message it can't process before it gets the `DoneCommand`. But if, for some reason, neither the event or commands gets delivered, it would time out and stay green. So I think I should assert on done being true here.